### ISSUE TITLE
Simulation stopped for unknown status

### ISSUE THREAD

**Issue by maple10001** [Azure/device-simulation-dotnet#156]
# Type of issue 

- [x] Bug
- [ ] New feature
- [ ] Enhancement

# Description 
Simulation stopped for `unknown` status
...

# Steps to reproduce 

1. Deploy a device simulation service with S3 hub, VM (16 cores, 64GB mem)
2. Run a simulation with over 1k devices
3. Simulation possible (happens 2 times out of 10 deployments) stopped after 30mins - 2hrs 

# Expected behavior 
- Simulation stopped with a reason 

# Current behavior 
...

# Possible solution 
...

# Context and Environment
* Operating System: ...
* GitHub branch: ...
* .NET Runtime: ...

# Code to reproduce the bug 
```
using Xunit;

public class MyTest
{
    [Fact]
    public void TestToReproduceIssue()
    {
        // Arrange
        ...

        // Act
        ...

        // Assert
        ...
    }
}
```

**Comment by dluc** [comment]
Is this still happening ?

**Comment by maple10001** [comment]
Fixed with PR #162 .
